src/lib.rs

```diff
@@ -38,6 +37,46 @@ fn get_args(current_dir: &OsStr) -> clap::ArgMatches {
         .get_matches()
 }
 
+fn run_checks(fe: &FileEntry) -> io::Result<Vec<Warning>> {
+    let f = File::open(fe.path.clone())?;
+    let mut reader = BufReader::new(f);
+
+    // TODO: Initialize a vector with a capacity equal to the number of lines
+    let mut lines: Vec<LineEntry> = Vec::new();
+
+    let mut number = 0;
+    loop {
```

Comment:
I don't like the idea of using `loop` to extract data from a file.
Previous way with `for (index, line) in reader.lines().enumerate() {}` worked fine:
```rust
// TODO: Initialize a vector with a capacity equal to the number of lines
let mut lines: Vec<LineEntry> = Vec::new();
let mut number = 0;
for (index, line) in reader.lines().enumerate() {
    number = index + 1;
    let raw_string = line?;

    lines.push(LineEntry {
        number,
        file_path: file.path.clone(),
        raw_string,
    })
}

let result = checks::run(lines);
warnings.extend(result);
```

You can get the last character from the file more easily:
```rust
// TODO: Initialize a vector with a capacity equal to the number of lines
let mut lines: Vec<LineEntry> = Vec::new();
let mut number = 0;
for (index, line) in reader.lines().enumerate() {
// ...
}

f.seek(SeekFrom::End(-1));
let mut reader = BufReader::new(&f);
let mut s = String::new();
reader.read_to_string(&mut s)?;

lines.push(LineEntry {
    number: number + 1,
    file_path: file.path.clone(),
    raw_string: s,
});

let result = checks::run(lines);
warnings.extend(result);
```

Reply:
Yes, I tryed this approach too, I have it in the separate branch locally.
I can easily move to it.

Reply:
Done